temporal-sdk/src/test/java/io/temporal/workflow/updateTest/UpdateTest.java

```diff
@@ -103,6 +107,55 @@ public void testUpdate() {
     assertEquals("Execute-Hello Update Execute-Hello Update 2", result);
   }
 
+  private static class FakesResultUpdateInterceptor extends WorkflowClientInterceptorBase {
+    @Override
+    public WorkflowClientCallsInterceptor workflowClientCallsInterceptor(
+        WorkflowClientCallsInterceptor next) {
+      return new WorkflowClientCallsInterceptorBase(next) {
+        @Override
+        public <R> UpdateHandle<R> startUpdate(StartUpdateInput<R> input) {
+          super.startUpdate(input);
```

Comment:
Side note: Why don't we have something like `InputType.fromExisting(input)` and then some way to relatively easily modify the input? It's a bit annoying to have to re-pass every field through to a new constructor if you want to modify inputs.

I get why they're not directly mutable, but, might be nice to have a convenient way to do that.

Reply:
Modify inputs is not really an encouraged pattern. Not that some users don't but I it is a advanced path to take.